00_homework_kawapa/Z9/1-template_matching/main.cpp

```diff
@@ -59,22 +55,27 @@ int main(int argc, char **argv)
     Mat img_display;
     img.copyTo(img_display);
 
-    std::vector<std::thread> threads;
+    std::vector<std::future<cv::Point>> futures(templates);
 
     for (int i = 0; i < templates; i++)
-        threads.emplace_back(std::thread(MatchingMethod, img, templ[i], std::ref(img_display), match_method));
+        futures[i] = std::async(std::launch::async, MatchingMethod, std::cref(img), std::cref(templ[i]), match_method);
     
-    for (auto &&i : threads)
-        i.join();
+    for (int i = 0; i < templates; i++)
+    {
+        auto tmp = futures[i].get();
+        rectangle(img_display, tmp, Point(tmp.x + templ[i].cols, tmp.y + templ[i].rows), Scalar::all(0), 2, 8, 0);
+    }
 
     imshow(image_window, img_display);
```

Comment:
nie wiem czy dobrze pamiętam, ale możesz sprawdzić taką ciekawą rzecz, czy jakbyś przeniósł imshow przed rysowanie prostokątów, czyli przed linię 63, to czy później ich dodawanie nie powoduje automatycznej aktualizacji rysunku. Wtedy fajnie można by było obserwować efekty wyszukiwań na bieżąco. Sprawdzisz? :)

Reply:
Sprawdziłem. Pokazuje się tylko oryginalne zdjęcie. Nie odświeża się w trybie rzeczywistym :(